cmake/modules/SwiftSharedCMakeConfig.cmake

```diff
@@ -226,7 +226,7 @@ macro(swift_common_unified_build_config product)
   set(PATH_TO_LLVM_BUILD "${CMAKE_BINARY_DIR}")
   set(${product}_PATH_TO_CLANG_BUILD "${CMAKE_BINARY_DIR}")
   set(PATH_TO_CLANG_BUILD "${CMAKE_BINARY_DIR}")
-  set(CLANG_MAIN_INCLUDE_DIR "${CMAKE_SOURCE_DIR}/tools/clang/include")
+  set(CLANG_MAIN_INCLUDE_DIR "${LLVM_EXTERNAL_CLANG_SOURCE_DIR}/include")
```

Comment:
One worry is what if someone assumes that clang/cmark aren't peers and tries to do a uniform build? Can you add some sort of warning/assert?

Reply:
Well, `LLVM_EXTERNAL_CLANG_SOURCE_DIR` and `LLVM_EXTERNAL_CMARK_SOURCE_DIR` are computed through LLVM's machinery and will point to the right location (alternatively, as Chris mentioned, a user can put that into the cache with force and cause that to be overridden).  This only sets up the include paths.  Is there some other invariant that Im missing?

What would you like me to add a `precondition` on?